specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,316 @@
+  /// (https://w3c.github.io/webappsec-secure-contexts#potentially-trustworthy-origin) 
+  /// a checkmark box will be displayed on the default browser UI that gives the user 
+  /// the option to always allow the external protocol to launch from this origin.
+  /// If the user checks this box, upon the next request from that origin to the 
+  /// protocol, the event will still be raised. 
+  ///
+  /// If the request is initiated by a cross-origin iframe without a user gesture, 
+  /// the request will be blocked and the `LaunchingExternalProtocol` event will not 
+  /// be raised. 
+  /// If the request is initiated by a browser popup window or in another case in 
+  /// which the embedded browser is null, the `LaunchingExternalProtocol` event will 
+  /// not be raised and the request will continue as normal and display the default browser
+  /// UI dialog to the user.
+  /// \snippet SettingsComponent.cpp LaunchingExternalProtocol
+  HRESULT add_LaunchingExternalProtocol(
+      [in] ICoreWebView2LaunchingExternalProtocolEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with
+  /// `add_LaunchingExternalProtocol`.
+  HRESULT remove_LaunchingExternalProtocol(
+      [in] EventRegistrationToken token);
+  }
+
+/// Receives the `LaunchingExternalProtocol` event.
+[uuid(e5fea648-79c9-47aa-8314-f471fe627649), object, pointer_default(unique)] 
+interface ICoreWebView2LaunchingExternalProtocolEventHandler: IUnknown {
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2* sender,
+      [in] ICoreWebView2LaunchingExternalProtocolEventArgs* args);
+}
+
+/// Event args for `LaunchingExternalProtocol` event.
+[uuid(fc43b557-9713-4a67-af8d-a76ef3a206e8), object, pointer_default(unique)] 
+interface ICoreWebView2LaunchingExternalProtocolEventArgs: IUnknown {
+  /// The URI of the requested external protocol.
+
+  [propget] HRESULT Uri([out, retval] LPWSTR* uri);
+
+  /// The origin initiating the external protocol launch.
+  /// The origin will be empty if the WebView2 navigates to the external protocol.
+
+  [propget] HRESULT InitiatingOrigin([out, retval] LPWSTR* uri);
+
+  /// `TRUE` when the external protocol request was initiated through a user gesture.
+  ///
+  /// \> [!NOTE]\n\> Being initiated through a user gesture does not mean that user intended
+  /// to access the associated resource.
+
+  [propget] HRESULT IsUserInitiated([out, retval] BOOL* isUserInitiated);
+
+  /// `TRUE` when the external protocol request was initiated via a non-main frame that 
+  /// has a different origin then the owning top-level page.
+
+  [propget] HRESULT IsCrossOriginIframe([out, retval] BOOL* isCrossOriginIframe);
+
+  /// The host may set this flag to cancel the external protocol launch.  If set to
+  /// `TRUE`, the external protocol will not be launched, and the default
+  /// dialog is not displayed.

```

Comment:
It would be good to note here how we expect the end dev to use this property to do custom handling

```suggestion
  /// dialog is not displayed. You can use this to manually replace the normal handling of launching external protocols
  /// to show your own dialog or apply your own security logic.
```